source/documentation/adrs/adr-008.html.md.erb

```diff
@@ -0,0 +1,85 @@
+---
+title: ADR-008 Deployment environments
+last_reviewed_on: 2023-11-27
+review_in: 12 months
+---
+# <%= current_page.data.title %>
+
+### Date: 27-11-2023
+
+## Status
+
+🤔 Proposed
+
+## Context
+
+When working across multiple teams, or switching between teams, it can be hard to follow what each deployment environment is for and what practices it keeps. This can lead to data ending up in an improper environment, or tests being run with incorrect assumptions of the state of the environment.
+
+We sometimes need to test, demo or explore functionality between multiple connected services. To do so, we need persistent environments of each service that are connected together. Not all products have this, and those that do can be inconsistent, making it harder to perform these activities.
+
+## Decision
+
+All services must have the following environments:
+
+- Production
+- Preproduction
+- Demo
+- Local
+
+Services may commonly have additional environments:
+
+- Automatic PR environments
+- Manual PR environments
+- Other persistent environments
+
+### Production environment
+
+This is what our users access and it contains our real data. It must is the only environment in the production account.
+
+### Preproduction environment
+
+This has identical infrastructure to the production environment and is the only environment in the preproduction account. It is not accessible by the public. It is deployed in the pipeline before production to test integration that the release can be deployed successfully.
+
+It is be available at `preproduction.{production host}`.
+
+### Demo environment
+
+This is used to test, demo and explore functionality against the currently deployed release. It is in the development account, does not contain production data and is not accessible by the public. It is deployed in the pipeline after production to ensure it reflects the latest releaase.
+
+It is connected to the demo environment of any dependencies, and is the target of the demo environemnt of any dependent services. This allows cross-service testing.
+
+It is be available at `demo.{production host}`.
+
```

Comment:
I would also add that a primary development environment that closely resembles the production infrastructure but is of a smaller scale should exist for all services to allow tests to be carried out on a prod-like real infra but without any real data.

I feel deployment pipelines should be Development Deployment -> Development Testing -> Preproduction Deployment -> Production Deployment.
Lighter weight "smoke-tests" that do not change or create data should be available to validate the service is usable at all times.

Reply:
See comment below, but I've generalised this a bit to separate out the needs of an in-pipeline production-infra environment and a maybe-out-of-pipeline production-data environment.